src/platform_impl/linux/mod.rs

```diff
@@ -34,175 +51,213 @@ const BACKEND_PREFERENCE_ENV_VAR: &str = "WINIT_UNIX_BACKEND";
 
 #[derive(Clone)]
 pub struct PlatformSpecificWindowBuilderAttributes {
+    #[cfg(feature = "x11")]
     pub visual_infos: Option<XVisualInfo>,
+    #[cfg(feature = "x11")]
     pub screen_id: Option<i32>,
+    #[cfg(feature = "x11")]
     pub resize_increments: Option<Size>,
+    #[cfg(feature = "x11")]
     pub base_size: Option<Size>,
+    #[cfg(feature = "x11")]
     pub class: Option<(String, String)>,
+    #[cfg(feature = "x11")]
     pub override_redirect: bool,
+    #[cfg(feature = "x11")]
     pub x11_window_types: Vec<XWindowType>,
+    #[cfg(feature = "x11")]
     pub gtk_theme_variant: Option<String>,
+    #[cfg(feature = "wayland")]
     pub app_id: Option<String>,
 }
 
 impl Default for PlatformSpecificWindowBuilderAttributes {
     fn default() -> Self {
         Self {
+            #[cfg(feature = "x11")]
             visual_infos: None,
+            #[cfg(feature = "x11")]
             screen_id: None,
+            #[cfg(feature = "x11")]
             resize_increments: None,
+            #[cfg(feature = "x11")]
             base_size: None,
+            #[cfg(feature = "x11")]
             class: None,
+            #[cfg(feature = "x11")]
             override_redirect: false,
+            #[cfg(feature = "x11")]
             x11_window_types: vec![XWindowType::Normal],
+            #[cfg(feature = "x11")]
             gtk_theme_variant: None,
+            #[cfg(feature = "wayland")]
             app_id: None,
         }
     }
 }
 
+#[cfg(feature = "x11")]
 lazy_static! {
     pub static ref X11_BACKEND: Mutex<Result<Arc<XConnection>, XNotSupported>> =
         Mutex::new(XConnection::new(Some(x_error_callback)).map(Arc::new));
 }
 
 #[derive(Debug, Clone)]
 pub enum OsError {
+    #[cfg(feature = "x11")]
     XError(XError),
+    #[cfg(feature = "x11")]
     XMisc(&'static str),
 }
 
 impl fmt::Display for OsError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+    fn fmt(&self, _f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        #[cfg(not(feature = "x11"))]
+        panic!();
```

Comment:
It should be possible to remove this `panic!()` and simply mark each `match` arm below with `#[cfg(feature = "x11")]`. It may be counter-intuitive, but the construct `match foo {}` is valid for an `enum` value with no defined variants.

Reply:
I thought the same, but it did not work.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=482ef496084d4ac4ae03268489e6108f

Reply:
Actually, i managed to make it work by dereferencing the value